main/main.go

```diff
@@ -542,46 +550,39 @@ func (difftool *xdcrDiffTool) waitForDuration(sourceDcpDriver, targetDcpDriver *
 
 func (difftool *xdcrDiffTool) retrieveReplicationSpecInfo() error {
 	// CBAUTH has already been setup
-	rcMap, err := difftool.remoteClusterSvc.RemoteClusters()
+	var err error
+	difftool.specifiedRef, err = difftool.remoteClusterSvc.RemoteClusterByRefName(options.remoteClusterName, false /*refresh*/)
 	if err != nil {
 		difftool.logger.Errorf("Error retrieving remote clusters: %v\n", err)
 		return err
 	}
 
+	if options.targetUsername != "" && options.targetPassword != "" {
+		difftool.logger.Infof("Target username and password were specified. Overriding them in the retrieved remote cluster reference")
+		difftool.specifiedRef.UserName_ = options.targetUsername
+		difftool.specifiedRef.Password_ = options.targetPassword
+	}
+
 	specMap, err := difftool.replicationSpecSvc.AllReplicationSpecs()
 	if err != nil {
 		difftool.logger.Errorf("Error retrieving specs: %v\n", err)
-	}
-
-	for _, ref := range rcMap {
-		if ref.Name_ == options.remoteClusterName {
-			difftool.specifiedRef = ref
-			break
-		}
+		return err
 	}
 
 	for _, spec := range specMap {
-		if spec.SourceBucketName == options.sourceBucketName && spec.TargetBucketName == options.targetBucketName {
+		if spec.SourceBucketName == options.sourceBucketName && spec.TargetBucketName == options.targetBucketName && spec.TargetClusterUUID == difftool.specifiedRef.Uuid() {
 			difftool.specifiedSpec = spec
 			break
 		}
 	}
 
-	var errStrs []string
-	if difftool.specifiedRef == nil {
-		errStrs = append(errStrs, fmt.Sprintf("Unable to find Remote cluster %v\n", options.remoteClusterName))
-	}
 	if difftool.specifiedSpec == nil {
-		errStrs = append(errStrs, fmt.Sprintf("Unable to find Replication Spec with source %v target %v\n", options.sourceBucketName, options.targetBucketName))
-	}
-	if len(errStrs) > 0 {
-		err := fmt.Errorf(strings.Join(errStrs, " and "))
+		err = fmt.Errorf("Unable to find Replication Spec with source %v target %v\n", options.sourceBucketName, options.targetBucketName)
 		difftool.logger.Errorf(err.Error())
 		return err
```

Comment:
not having a spec may not necessarily be an error condition. if customer simply wants to compare two clusters, with repl spec between them already deleted, we should let them.

Reply:
Sure.